sources/Launcher/DYNCriticalTimeLauncher.cpp

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source suite
+// of simulation tools for power systems.
+//
+
+/**
+ * @file  CriticalTimeLauncher.cpp
+ *
+ * @brief Critical Time launcher: implementation of the algorithm and interaction with dynawo core
+ *
+ */
+
+#ifdef WITH_OPENMP
+#include <omp.h>
+#endif
+
+#include <DYNExecUtils.h>
+#include <DYNSimulation.h>
+#include <DYNSubModel.h>
+#include <DYNModelMulti.h>
+#include "DYNCriticalTimeLauncher.h"
+#include "DYNMultipleJobs.h"
+#include "DYNAggrResXmlExporter.h"
+#include "MacrosMessage.h"
+
+using multipleJobs::MultipleJobs;
+
+namespace DYNAlgorithms {
+
+void
+CriticalTimeLauncher::launch() {
+  const std::string& jobsFile = criticalTimeCalculation_->getJobsFile();
+
+#ifdef WITH_OPENMP
+omp_set_num_threads(nbThreads_);
+#endif
+
+  inputs_.readInputs(workingDirectory_, jobsFile, 1);
+  boost::shared_ptr<job::JobEntry> job = inputs_.cloneJobEntry();
+  SimulationParameters params;
+  boost::shared_ptr<DYN::Simulation> simulation = createAndInitSimulation(workingDirectory_, job, params, results_, inputs_);
+  
+  if (simulation) {
+    boost::shared_ptr<DYN::ModelMulti> modelMulti = boost::dynamic_pointer_cast<DYN::ModelMulti>(simulation->model_);
+    std::string DDBDir = getEnvVar("DYNAWO_DDB_DIR");
+
+    const std::string& dydId = criticalTimeCalculation_->getDydId();
+    const std::string& endPar = criticalTimeCalculation_->getEndPar();
+
+    subModel_ = modelMulti->findSubModelByName(dydId);
+    subModel_->setParameterValue(endPar, DYN::PAR, tSup_, false);
```

Comment:
I don't see any usage of startPar?

Reply:
Indeed I don't use it. I removed all appearances of this useless variable.